### PR TITLE
SQL 1031: Add SQLTables edge case handling

### PR DESCRIPTION
**Pull request by bucaojit**
Added MongoTableTypes struct to handle calls for `SQL_ALL_TABLE_TYPES`. 
Returning `ColIndexOutOfBounds` error when index is out of bounds. 
Added empty result response to SQL_ALL_SCHEMAS.

### CODE REVIEW

**Review by nbagnard — CHANGES_REQUESTED** [review]
Looking pretty good. I only have 2 comments and one of them being a small optimization.

File: core/src/table_types.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    table_type: Vec<String>,
+    // The current table type index
+    current_table_type_index: usize,
+}
+
+impl MongoTableTypes {
+    // Statement for SQLTables("", "", "", SQL_ALL_TABLE_TYPES ).
+    pub fn all_table_types() -> MongoTableTypes {
+        MongoTableTypes {
+            table_type: vec!["TABLE".to_string(), "VIEW".to_string()],
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply core/src/table_types.rs]: Sure, added the static const array.

File: core/src/databases.rs [hunk 2]
```diff
@@ -234,19 +235,16 @@ impl MongoStatement for MongoDatabases {
     fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply core/src/databases.rs:235]: I added the `row_index_is_valid` attribute in StatementHandle which gets set to true/false depending on the result of SQLFetch()->statement.next().  It gets checked in SQLGetData() before statement.get_value() and if the row index is not valid returns `InvalidCursorState`.  Let me know what you think.

File: core/src/table_types.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nbagnard) [reply core/src/table_types.rs]: 👍

File: core/src/databases.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbagnard) [reply core/src/databases.rs:235]: Yes, it works :)

**Review by nbagnard — APPROVED** [review]
LGTM! 🎉

**Review by pmeredit — APPROVED** [review]
LGTM!

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**